packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -77,6 +81,7 @@ export function loadConfigAndTasks(
 
   if (showSolidityConfigWarnings) {
     checkMissingSolidityConfig(userConfig);
+    checkUnsupportedRemappings(userConfig);
```

Comment:
@alcuadrado Would it be a bit better if this was done in the next `if (showSolidityConfigWarnings` (when the config is already resolved)? It might be a stretch, but maybe some plugin adds it or something. Or do you prefer to just check what users write?

Reply:
Plus, that should simplify the logic of the check function a lot.